[Home] - CSS 모듈화

kelvin-chihyun opened the pull request:
## 🔍️ 이 PR을 통해 해결하려는 문제가 무엇인가요?

> 어떤 기능을 구현한건지, 이슈 대응이라면 어떤 이슈인지 PR이 열리게 된 계기와 목적을 Reviewer 들이 쉽게 이해할 수 있도록 적어 주세요
> 일감 백로그 링크나 다이어그램, 피그마를 첨부해도 좋아요

- 메인 페이지의 컴포넌트들을 CSS 모듈화하였습니다.
- 디자인 시안과 일치하지 않는 일부 CSS를 수정하였습니다.

## ✨ 이 PR에서 핵심적으로 변경된 사항은 무엇일까요?

> 문제를 해결하면서 주요하게 변경된 사항들을 적어 주세요

-

## 🔖 핵심 변경 사항 외에 추가적으로 변경된 부분이 있나요?

> 없으면 "없음" 이라고 기재해 주세요

-

## 🙏 Reviewer 분들이 이런 부분을 신경써서 봐 주시면 좋겠어요

> 개발 과정에서 다른 분들의 의견은 어떠한지 궁금했거나 크로스 체크가 필요하다고 느껴진 코드가 있다면 남겨주세요

-

## 🩺 이 PR에서 테스트 혹은 검증이 필요한 부분이 있을까요?

> 테스트가 필요한 항목이나 테스트 코드가 추가되었다면 함께 적어주세요

-

### 📌 PR 진행 시 이러한 점들을 참고해 주세요

- Reviewer 분들은 코드 리뷰 시 좋은 코드의 방향을 제시하되, 코드 수정을 강제하지 말아 주세요.
- Reviewer 분들은 좋은 코드를 발견한 경우, 칭찬과 격려를 아끼지 말아 주세요.
- Review는 특수한 케이스가 아니면 Reviewer로 지정된 시점 기준으로 3일 이내에 진행해 주세요.
- Comment 작성 시 Prefix로 P1, P2, P3 를 적어 주시면 Assignee가 보다 명확하게 Comment에 대해 대응할 수 있어요
  - P1 : 꼭 반영해 주세요 (Request Changes) - 이슈가 발생하거나 취약점이 발견되는 케이스 등
  - P2 : 반영을 적극적으로 고려해 주시면 좋을 것 같아요 (Comment)
  - P3 : 이런 방법도 있을 것 같아요~ 등의 사소한 의견입니다 (Chore)

#

---

## 📝 Assignee를 위한 CheckList

- [ ] To-Do Item

kelvin-chihyun committed:
🌈 [Design] : 디자인 일부 수정

반응형으로 모바일 레이아웃일때, flex로 두면 일부 텍스트가 가독성이 좋지 않아
모바일 환경에서 flex-col로 두었습니다.

kelvin-chihyun committed:
🐛 [Fix] : export 이중 정의 오류 수정

export가 두번 정의되어, dev 환경에서는 정상적으로 작동하지만,
eslint 상에 오류가 발생하여 수정하였습니다.

kelvin-chihyun committed:
🤝 [Merge] : 원격 저장소와 로컬 저장소 동기화

kelvin-chihyun committed:
➕ [Add] : 이미지 파일 추가

메인 페이지 - Products 섹션에 들어가는
이미지 파일을 추가하였습니다.

kelvin-chihyun committed:
⚙️ [Chore] : tawilwind.config.js 수정

병합되면서 날아간 코드를 추가하였습니다.
'--open-accent-accent-1': '#3fa9f5',
'--open-accent-accent-2': '#0b6399',

kelvin-chihyun committed:
🌈 [Design] : hover 시 효과

디자인 시안 상 있는 hover 효과가 누락되어 추가하였습니다.

kelvin-chihyun committed:
🌈 [Design] : 폰트 수정

디자인 시안과 불일치하는 폰트를 수정하였습니다.

kelvin-chihyun committed:
♻️ [Refactor] : CSS 모듈화

메인 페이지의 컴포넌트들을 모듈화 하였습니다.

kelvin-chihyun committed:
🌈 [Design] : 태그 캔버스 css 수정

태그 캔버스의 height가 길어서,
height를 700 => 550으로 조정하였습니다.

kelvin-chihyun committed:
➖ [Remove] : .env.production 파일 삭제

초기 설정 오류로 .env.production 파일이 원격 저장소에
포함되어서, 삭제합니다.

kelvin-chihyun committed:
♻️ [Refactor]  : index.html 언어 변경

<html lang="en">으로 초기 설정하여,
<html lang="ko">로 수정했습니다.

kelvin-chihyun committed:
🛡️ [Security] : .gitignore 파일 수정

.gitignore 파일 내 .env.production을 적었으나,
인식되지 않아 *.production으로 수정하였습니다.

kelvin-chihyun committed:
➖ [Remove] : 기존 favicon 파일 삭제

기존 favicon 파일을 삭제하였습니다.

kelvin-chihyun committed:
➕ [Add]	: favicon 파일 추가

새로운 favicon 파일을 추가하였습니다.

kelvin-chihyun committed:
⚙️ [Chore] : favicon 파일 변경

favicon 파일을 변경하였습니다.

kelvin-chihyun committed:
🌈 [Design] : 레이아웃 변경

Solutions 컴포넌트 레이아웃이 일부 깨지는 현상이 있어,
width와 height를 수정하였습니다.

Related to #94

kelvin-chihyun committed:
🏛️ [Style] : 모듈 클래스명 변경

동일 클래스명 지정으로 CSS가 덮어씌워지는 현상이 발생하여
클래스명을 변경하였습니다.

Related to #94

ryudg reviewed (approved):
이상 없습니다!!

dani-day commented on src/components/Home/Home.module.css:
```diff
@@ -1,3 +1,242 @@
 .swiper-wrapper > div {
   @apply relative ease-linear;
 }
```
P1: 치현님 이 부분 `.partnerSwiper > div` 로 두어야 적용되는 것 아닌지 확인 부탁드리겠습니다. 정지/시작 속도가 아직 남아있는 걸로 보입니다.

kelvin-chihyun replied:
넵 수정했습니다!

dani-day commented on src/components/Home/Home.module.css:
```diff
@@ -1,3 +1,242 @@
+      @apply tablet:gap-open-2xl desktop:gap-open-2xl;
+
+      & .cardWrapper {
+        @apply w-full flex justify-between items-center gap-open-4xl p-open-2xl  -bg--open-gray-950 rounded-[30px] border  border-transparent;
```
P1: cardWrapper 의 border 가 시안에서는 4px 인데 현재 1px 로 적용된 것으로 보입니다. 다만 확인해보니 4px 은 실제 구현 시에 다소 두꺼워보여서 2px 로 수정해주시면 감사하겠습니다 :)

kelvin-chihyun replied:
수정 완료했습니다!

dani-day commented on src/components/Home/Home.module.css:
```diff
@@ -1,3 +1,242 @@
+      }
+    }
+
+    & .industriesContainer {
+      @apply w-full flex flex-wrap flex-col justify-center rounded-l-[30px] overflow-hidden;
```
P2: overflow-visible 로 두는 것은 어떤지 제안드립니다:)
현재 산업군 카드 hover 시 컨테이너 바깥으로 나가는 부분이 잘려보여 확인해보니 industriesContainer 에 overflow-visible 을 주면 해결되는 것으로 보입니다.

![스크린샷 2023-12-18 오후 4 55 50](https://github.com/ReOpenSG/ReOpenSG/assets/134567470/22d87537-4f89-4f57-a0dc-9f28880b3033)

kelvin-chihyun replied:
수정 완료했습니다!

dani-day commented on src/components/Home/Home.module.css:
```diff
@@ -1,3 +1,242 @@
+    & .industriesContainer {
+      @apply w-full flex flex-wrap flex-col justify-center rounded-l-[30px] overflow-hidden;
+
+      & .cardWrapper {
```
P3: 디자인 시안 상, cardWrapper 의 높이가 3개 모두 동일하게 변경하는 것을 제안드립니다. 현재 피그마 시안에서는 타블렛, PC 의 경우 height 값이 400 으로 고정되어있으므로, 이 점 고려 부탁드립니다.

kelvin-chihyun replied:
수정하였습니다!

kelvin-chihyun committed:
🐛 [Fix] : 라우팅 추가

Solutions 컴포넌트 내 Learn More 버튼 클릭시,
각 솔루션의 상세페이지로 이동하도록 라우팅을 추가하였습니다.

Related to #94

dani-day commented on src/components/Home/Products.jsx:
```diff
@@ -9,36 +13,33 @@ function Products() {
       name: 'Open Solutions',
       desc: 'OpenSG의 solution을 경험해보세요.',
       route: '/solutions/ACS',
+      imgUrl: Product1,
     },
     {
       id: 2,
       name: 'Open Machines',
       desc: 'OpenSG의 machine을 경험해보세요.',
       route: '/machines/Lift Type',
+      imgUrl: Product2,
     },
     {
       id: 3,
       name: 'Open Devices',
       desc: 'OpenSG의 device를 경험해보세요.',
       route: '/devices/Microsoft',
+      imgUrl: Product3,
```
P1: Solutions, Smart Machine, Smart Device 로 시안과 동일하게 변경 부탁드립니다.

kelvin-chihyun replied:
수정하였습니다!

dani-day reviewed (changes requested):
남겨둔 의견 확인 부탁드리겠습니다 👍

kelvin-chihyun committed:
🐛 [Fix] : 스와이퍼 관련 CSS 일부 수정

PR Comment에 따라스와이퍼에 timing function이 적용되지 않아,
정지/시작 시 부자연스러웠던 효과를 수정하였습니다.

Related to #94

kelvin-chihyun committed:
🐛 [Fix] : border width 조정

PR comment에 따라,
Solutions 컴포넌트 내 Card의 border가 디자인 시안과 달라
border width를 1px에서 2px로 조정하였습니다.

Related to #94

kelvin-chihyun committed:
🐛 [Fix] : 이미지 잘림 현상 해결

PR comment에 따라, overflow- hidden처리되어 hover시 이미지가 잘리던 현상을
overflow: visible로 변경하여 해결하였습니다.

Related to #94

kelvin-chihyun committed:
🐛 [Fix] : Card width, height 수정

PR comment에 따라, Products 컴포넌트 내 cardWrapper의 높이를
tablet, desktop에서 400px로 고정하고, 모바일 레이아웃에서 width를
flex-1로 변경하였습니다.

Related to #94

kelvin-chihyun committed:
🐛 [Fix] : 명칭 수정

Card의 명칭을 변경하였습니다.
Open Solutions => Solutions,
Open Machines => Smart Machine,
Open Devices => Smart Devices

Related to #94

dani-day reviewed (approved):
수정사항 확인했고, 추가 의견 없습니다. :) 치현님 덕분에 메인페이지가 금방 완성되고 있네요. 수고하셨습니다! 👍
